tests/blinds/test_blindsschedule.py

```diff
@@ -50,4 +51,213 @@ def test_constructor( self ):
+                ScheduleTimeBlock( datetime.time( 23, 38), datetime.time( 23, 59 ), BlindMode.GREEN, None )
+            ],
+            BlindsSchedule.SATURDAY: []
+        }
+
+        blindsSchedule._schedule = sched
+        assert( blindsSchedule.checkHasNoTimeConflicts() == True )
+
+        # add time conflict
+        sched = {
+            BlindsSchedule.SUNDAY: [],
+            BlindsSchedule.MONDAY : [],
+            BlindsSchedule.TUESDAY: [],
+            BlindsSchedule.WEDNESDAY: [],
+            BlindsSchedule.THURSDAY: [],
+            BlindsSchedule.FRIDAY: [
+                ScheduleTimeBlock( datetime.time( 00, 32), datetime.time( 20, 00 ), BlindMode.LIGHT, None ),
+                ScheduleTimeBlock( datetime.time( 16, 44), datetime.time(23, 00 ), BlindMode.LIGHT, None ), 
+                ScheduleTimeBlock( datetime.time( 23, 38), datetime.time( 23, 59 ), BlindMode.GREEN, None )
+            ],
+            BlindsSchedule.SATURDAY: []
+        }
+
+        blindsSchedule._schedule = sched
+        with pytest.raises( BlindSchedulingException ):
+            blindsSchedule.checkHasNoTimeConflicts()
+
+    '''
+    Test for BlindSchedule serialization with toJson
+    '''
+    def test_serialization( self ):
+        default_mode = BlindMode.LIGHT
+        default_pos = None
+        sched = {
+            BlindsSchedule.SUNDAY: [
+                ScheduleTimeBlock( datetime.time( 23, 38), datetime.time( 23, 59 ), BlindMode.LIGHT, None ), 
+                ScheduleTimeBlock( datetime.time( 5, 00), datetime.time( 7, 00 ), BlindMode.MANUAL, 15 )
+            ],
+            BlindsSchedule.MONDAY : [ 
+                ScheduleTimeBlock( datetime.time( 12, 00), datetime.time( 15, 00 ), BlindMode.MANUAL, -33 ),
+                ScheduleTimeBlock( datetime.time( 4, 3), datetime.time( 6, 00 ), BlindMode.LIGHT, None )
+            ],
+            BlindsSchedule.TUESDAY: [
+                ScheduleTimeBlock( datetime.time( 10, 22), datetime.time( 12, 00 ), BlindMode.DARK, None )
+            ],
+            BlindsSchedule.WEDNESDAY: [],
+            BlindsSchedule.THURSDAY: [],
+            BlindsSchedule.FRIDAY: [
+                ScheduleTimeBlock( datetime.time( 22, 44), datetime.time(23, 00 ), BlindMode.LIGHT, None ), 
+                ScheduleTimeBlock( datetime.time( 23, 38), datetime.time( 23, 59 ), BlindMode.GREEN, None ), 
+                ScheduleTimeBlock( datetime.time( 00, 32), datetime.time( 1, 00 ), BlindMode.LIGHT, None )
+            ],
+            BlindsSchedule.SATURDAY: []
+        }
+
+        blindsSchedule = BlindsSchedule( default_mode, default_pos, sched )
+
+        # use the sortKeys flag to allow deterministic output for comparison
+        scheduleJson = BlindsSchedule.toJson( blindsSchedule, sortKeys=True )
+
+        # open schedule1.json to compare 
+        with open( "schedule1.json", "r" ) as file:
```

Comment:
Hmm, it seems like lines like these causes test failures with the CI system due to FileNotFound errors. Maybe it would be better to find the current directory of this file and then use the absolute path of the schedule files

Reply:
Fixed it now